src/scandeval/generation.py

```diff
@@ -321,6 +333,86 @@ def generate_single_iteration(
     return itr_scores
 
 
+def get_prefix_allowed_fn(
+    dataset_config: DatasetConfig, tokenizer: Tokenizer
+) -> Callable[[int, torch.Tensor], torch.Tensor] | None:
+    """Return the prefix allowed function to use for structured generation.
+
+    Args:
+        dataset_config:
+            The dataset config.
+        tokenizer:
+            The tokenizer.
+
+    Returns:
+        The prefix allowed function.
+    """
+    prefix_allowed_tokens_fns = list()
+
+    if dataset_config.task == NER and isinstance(tokenizer, PreTrainedTokenizer):
+        parser = get_ner_parser(dataset_config=dataset_config)
+        json_prefix_allowed_tokens_fn = build_transformers_prefix_allowed_tokens_fn(
+            tokenizer_data=tokenizer, character_level_parser=parser
+        )
+
+        # Do not allow newlines or tabs in the generated text
+        forbidden_token_ids = list(
+            set(list(tokenizer("\n\n\n\t\t\t", add_special_tokens=False).input_ids))
+        )
```

Comment:
it is because double newline can have its own token? Might this not break if e.g. a tokenizer has a custom token for triple newline? Might be worth spelling making clear which tokens are being forbidden.


E.g.
```
       forbidden_tokens = ["\n", "\n\n", "/t", "\t\t"]
        forbidden_token_ids = ...
```

Reply:
Changed it now.